### PR TITLE
NDoshi Classification

### PR DESCRIPTION
**Pull request by ndoshi83**

### CODE REVIEW

**Comment by axrd** [comment]
# Module 01 - Logistic Regression & One-Hot Encoding  

**Overall**: Well done! You're accuracy score is pretty darn good 👍 
**Feedback**: Amazing job commenting your code. Keep it up!

## Exploring/Cleaning data
- Try the `df.info()` method. Might help you as an initial step.
- Here's a helper function to help make checking for missing values functional (see what I did there? hehe):
```
def missing_values(data):
   total = data.isnull().sum()
   percent = data.isnull().sum().sort_values(ascending = False)/len(data)*100
   tp = pd.concat([total, percent], axis=1, keys=['Total', '%'])
   return(np.transpose(tp))
```
- But remember it's not just Nulls and NaNs. Read the links in the top of the lecture NB!

## Train/Val/Test Split
- 💯 / 💯 

## One-Hot Encoding & Scaling
- Nicely done! Kudos for evaluating cardinality of your features👍 

## Logistic Regression Model
 - Yep. That's how you do it 👍 

## Coefficients
- You need to retrieve your coefficients and plot them! 👎 
- In the Lecture notebook, Ryan H wrote a nifty way to do this:
```
# Import Matplotlib (inline)
%matplotlib inline
import matplotlib.pyplot as plt

coefficients = pd.Series(model.coef_[0], X_train_encoded.columns)
plt.figure(figsize=(10, 10))
coefficients.sort_values().plot.barh();
```

## Submit to Kaggle Competition
- Aww yeah! How's it feel to be on the leaderboard?

**Comment by axrd** [comment]
# Module 02 - Decision Trees  

**Overall**: Well done! Congratulations on breaking 80% Accuracy! 📈 
**Feedback**: Nicely done! Try implementing a `wrangle(X)` function like we did during lecture to help make your code more modular/functional. 

## More Exploring/Cleaning data
- Dig deeper. Lookout for missing or incorrect values in other Features. 
- Implement a wrangle function 

## Implement a Decision Tree Classifier
- Yep. Nicely done 👍 + 💯 for going ahead and implementing a forest!
- Plot the feature importances to help guide your approach
```
# Get feature importances
encoder = pipeline.named_steps['onehotencoder']
tree = pipeline.named_steps['decisiontreeclassifier']
feature_names = encoder.transform(X_val).columns
importances = pd.Series(tree.feature_importances_, feature_names)

# Plot feature importances
n = 20
plt.figure(figsize=(10,n/2))
plt.title(f'Top {n} features')
importances.sort_values()[-n:].plot.barh(color='grey');
```

## Submit New Predictions to Kaggle
- I see you climbing the leaderboard 🙌

**Review by brennerhaverlock — CHANGES_REQUESTED** [review]
# Nayan Nicolae- Logistic_Regression_Categorical_Encoding


# Part 1. Imports

***
**What went well:**

-Good collective use of imports:

```python

# Import all required libraries
%matplotlib inline
import category_encoders as ce
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import seaborn as sns
from sklearn.impute import SimpleImputer
from sklearn.model_selection import train_test_split
from sklearn.pipeline import make_pipeline
from sklearn.preprocessing import RobustScaler
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
pd.set_option('display.float_format', '{:.2f}'.format)
pd.set_option('display.max_columns', 50)

```

-IMPRESSIVE MERGE :) and comment : 

```python

# Import data from sources
train = pd.merge(pd.read_csv('https://drive.google.com/uc?export=download&id=14ULvX0uOgftTB2s97uS8lIx1nHGQIB0P'),
                 pd.read_csv('https://drive.google.com/uc?export=download&id=1r441wLr7gKGHGLyPpKauvCuUOU556S2f'))
test = pd.read_csv('https://drive.google.com/uc?export=download&id=1wvsYl9hbRbZuIuoaLWCsW_kbcxCdocHz')
submission = pd.read_csv('https://drive.google.com/uc?export=download&id=1kfJewnmhowpUo381oSn3XqsQ6Eto23XV')
train.shape, test.shape, submission.shape

```
**PCA FEATURES**

-
-Great use of [.describe](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.describe.html) to analyze both numeric and object series:

```python
print(train.describe())
print(train.describe(exclude='number'))
train_features.select_dtypes(exclude='number').nunique()

```


# What to work on
- [x] Comment more in code
- [x] Come to PM for questions or help
